File: rb/features.rb
# Ipleak SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module IpleakFeatures
  def self.make_feature(name)
    case name
    when "base"
      IpleakBaseFeature.new
    when "test"
      IpleakTestFeature.new
    else
      IpleakBaseFeature.new
    end
  end
end

File: py/ipleak_sdk/features.py
# Ipleak SDK feature factory

from ipleak_sdk.feature.base_feature import IpleakBaseFeature
from ipleak_sdk.feature.test_feature import IpleakTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IpleakBaseFeature(),
        "test": lambda: IpleakTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
